File: packages/java/parser-jvm-plugin-nonnull/src/test/java/dev/hilla/parser/plugins/nonnull/superclassmethods/PersonEndpoint.java
package dev.hilla.parser.plugins.nonnull.superclassmethods;

@Endpoint
public class PersonEndpoint extends CrudEndpoint<PersonEndpoint.Person, Integer>
        implements PagedData<PersonEndpoint.Person> {

    public static class Person {
        private String name;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }
}

File: packages/java/parser/src/test/kotlin/com/vaadin/hilla/parser/plugins/nonnull/kotlin/superclasses/endpointexposed/PersonEndpoint.kt
package com.vaadin.hilla.parser.plugins.nonnull.kotlin.superclasses.endpointexposed

import com.vaadin.hilla.parser.plugins.nonnull.kotlin.annotation.Endpoint

@Endpoint
class PersonEndpoint : CrudEndpoint<PersonEndpoint.Person, Int>(), PagedData<PersonEndpoint.Person> {

    class Person {
        var name: String? = null
    }

    fun create(entity: Person): Int {
        return 0
    }
}
